### DISCUSSION TITLE
Performance of Hybrid Rendering

### DISCUSSION THREAD

**Discussion by ghost** [master-co/css#35] [Question]
Hi, I'm a user of TailwindCSS and I have found just this fantastic alternative: Master CSS

Performance is always my main concern. As far as I know, the JIT engine of TailwindCSS is ~355kB (no compression) and it is too heavy to be loaded for every request.

I saw that Master CSS is also using the JIT engine? Is it using the same technique as TailwindCSS does? Would it slow or affect the performance of a large project?

**Comment by JacobLinCool** [comment]
Master CSS has its own JIT engine with optimized parsing algorithm, and it has no external dependency.

By contrast, Tailwind is more heavy than Master CSS because it is built on the top of PostCSS. 

As far as I know, Tailwind has improved its JIT engine, and now [it is even faster than WindiCSS](https://github.com/unocss/unocss#benchmark). But it is still slower than other solutions like Master CSS because runtime parsing is not what it focused on.
